src/app/leaves/add/components/LeaveRequestFormBuilder.tsx

```diff
@@ -96,16 +108,72 @@ export const LeaveRequestFormBuilder = () => {
   } = form;
 
   function onSubmit(values: z.infer<typeof leaveRequestSchema>) {
-    createLeaveRequest(values);
+    try {
+      if (!paramsId) {
+        createLeaveRequest(values);
+        toast({
+          title: "Success",
+          description: "Leave Request Created Successfully",
+        });
+      } else {
+        updateLeaveRequest(Number(paramsId), values);
+        toast({
+          title: "Success",
+          description: "Leave Request Updated Successfully",
+        });
+      }
+      router.push("/leaves");
+    } catch (error) {
+      toast({
+        title: "Error",
+        description: "Something went wrong! Please try again.",
+      });
+    }
   }
+  useEffect(() => {
+    const getLeaveById = async () => {
```

Comment:
This is wrong. See how @Mubashar1009 got this from server component.

Reply:
I have  Fixed this by following @Mubashar1009 pattern